src/Service/UserService.php

```diff
@@ -47,11 +47,38 @@ public function addUser(ObjectManager $entityManager, User $user): ?User
         return $user;
     }
 
-    public function bindUser(ObjectManager $entityManager, User $user, Customer $customer): void
+    public function bindingUser(ObjectManager $entityManager, User $user, Customer $customer = null): void
     {
         $user->setCustomer($customer);
 
-        $entityManager->persist($customer);
+        $entityManager->persist($user);
+        $entityManager->flush();
+    }
+
+    public function editUser(ObjectManager $entityManager, User $user, Customer $customer, string $currentPassword): void
```

Comment:
Passer deux param seulement:
1- userToUpdate = $this->getUser()
2- UserUpdateData = $requete->serialize(...) -> UserUpdateData: class qui contient que les champs qu'on veut modifier  comme firstname, lastname, email, password, username (sans customer, createdAt)

tu met à jour que les champs concernés (sans customer)
tu save $userToUpdate

Reply:
ObjectManager $entityManager à injecter dans le construstor